ksp_plugin_adapter/flight_planner.cs

```diff
@@ -406,6 +403,14 @@ class FlightPlanner : VesselSupervisedWindowRenderer {
     }
   }
 
+  internal void ModifyIthBurn(string vessel_guid, int i, Burn burn) {
```

Comment:
The class `FlightPlanner` is associated with a vessel (it is a `VesselSupervisedWindowRenderer`, so we should not pass a vessel—that would be an easy way to mess with the wrong vessel).
Use `predicted_vessel?.id.ToString()` instead.

The name `Ith` is weird unless you use named parameters, which you do not.
Rename to `ReplaceBurn(burn, i)`, so it looks like `FlightPlanReplace` (which it wraps, with additional housekeeping to keep the UI in sync).